src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1140,6 +1230,123 @@ private void IssueBuildSubmissionToScheduler(BuildSubmission submission, bool al
             }
         }
 
+        private void ExecuteGraphBuildScheduler(BuildSubmission submission)
+        {
+            try
+            {
+                if (_shuttingDown)
+                {
+                    throw new BuildAbortedException();
+                }
+
+                var projectGraph = submission.BuildRequestData.ProjectGraph;
+                if (projectGraph == null)
+                {
+                    // TODO: Need to pass through other stuff? Look how RequestBuilder creates ProjectInstance
```

Comment:
I'm not sure I understand what this means, can you elaborate?

Reply:
RequestBuilder creates project instances like this, which is way different than how ProjectGraph creates them:

```cs
            return new ProjectInstance(
                _requestEntry.RequestConfiguration.ProjectFullPath,
                globalProperties,
                toolsVersionOverride,
                _componentHost.BuildParameters,
                _nodeLoggingContext.LoggingService,
                new BuildEventContext(
                    _requestEntry.Request.BuildEventContext.SubmissionId,
                    _nodeLoggingContext.BuildEventContext.NodeId,
                    BuildEventContext.InvalidEvaluationId,
                    BuildEventContext.InvalidProjectInstanceId,
                    BuildEventContext.InvalidProjectContextId,
                    BuildEventContext.InvalidTargetId,
                    BuildEventContext.InvalidTaskId),
                sdkResolverService,
                _requestEntry.Request.SubmissionId,
                projectLoadSettings);

```

With Erik's recent change though I think I can more or less make this happen.